### ISSUE TITLE
infra: `make generate` is broken on ubuntu18

### ISSUE THREAD

**Issue by zirain** [envoyproxy/gateway#339]
make generate is broken since #193 

cc @danehans 

```
git checkout f7572928d42f8838c68c5133ab6252d01a93dfb8
make generate
```

```console
tools/bin/controller-gen object paths="./..."
panic: interface conversion: ast.Expr is *ast.CompositeLit, not *ast.Ident

goroutine 1403 [running]:
sigs.k8s.io/controller-tools/pkg/loader.(*referenceCollector).Visit(0xc003c83550, {0xd5b798?, 0xc00b8b5a70})
        /root/go/pkg/mod/sigs.k8s.io/controller-tools@v0.8.0/pkg/loader/refs.go:136 +0xff
go/ast.Walk({0xd591a0?, 0xc003c83550?}, {0xd5b798?, 0xc00b8b5a70?})
        /usr/local/go/src/go/ast/walk.go:52 +0x62
go/ast.walkExprList({0xd591a0, 0xc003c83550}, {0xc00b8c6610?, 0x1, 0xd5b838?})
        /usr/local/go/src/go/ast/walk.go:26 +0x91
go/ast.Walk({0xd591a0?, 0xc003c83550?}, {0xd5b1f8?, 0xc00b8cc980?})
        /usr/local/go/src/go/ast/walk.go:145 +0xa8b
go/ast.Walk({0xd591a0?, 0xc003c83550?}, {0xd5b180?, 0xc00b8c5710?})
        /usr/local/go/src/go/ast/walk.go:154 +0xfc5
go/ast.Walk({0xd591a0?, 0xc003c83550?}, {0xd5b090?, 0xc00b8c5740?})
        /usr/local/go/src/go/ast/walk.go:164 +0x1e49
go/ast.Walk({0xd591a0?, 0xc003c83550?}, {0xd5b3d8?, 0xc00b8cc9c0?})
        /usr/local/go/src/go/ast/walk.go:75 +0x37c
go/ast.Walk({0xd591a0?, 0xc003c83550?}, {0xd5b400?, 0xc00b8c5770?})
        /usr/local/go/src/go/ast/walk.go:86 +0x227a
go/ast.Walk({0xd591a0?, 0xc003c83550?}, {0xd5b838?, 0xc00b8b5a88?})
        /usr/local/go/src/go/ast/walk.go:169 +0x19a5
sigs.k8s.io/controller-tools/pkg/loader.(*referenceSet).collectReferences(0xc005ac6820, {0xd5e5b0?, 0xc00b8b5a88}, 0xc00b6dc600)
        /root/go/pkg/mod/sigs.k8s.io/controller-tools@v0.8.0/pkg/loader/refs.go:107 +0xd4
sigs.k8s.io/controller-tools/pkg/loader.allReferencedPackages.func1(0x203002?, 0xc009f5e920?, 0xc00b8cc880)
        /root/go/pkg/mod/sigs.k8s.io/controller-tools@v0.8.0/pkg/loader/refs.go:164 +0x52
sigs.k8s.io/controller-tools/pkg/loader.(*typeVisitor).Visit(0x0?, {0xd5b8b0?, 0xc00b8cc880?})
        /root/go/pkg/mod/sigs.k8s.io/controller-tools@v0.8.0/pkg/loader/visit.go:64 +0xe9
go/ast.Walk({0xd591c0?, 0xc00cc8f830?}, {0xd5b8b0?, 0xc00b8cc880?})
        /usr/local/go/src/go/ast/walk.go:52 +0x62
go/ast.Walk({0xd591c0?, 0xc00cc8f830?}, {0xd5b4f0?, 0xc00b8cca00?})
        /usr/local/go/src/go/ast/walk.go:345 +0x21ea
go/ast.walkDeclList({0xd591c0, 0xc00cc8f830}, {0xc00b69f100?, 0xd, 0x0?})
        /usr/local/go/src/go/ast/walk.go:38 +0x91
go/ast.Walk({0xd591c0?, 0xc00cc8f830?}, {0xd5b428?, 0xc00b8acd80?})
        /usr/local/go/src/go/ast/walk.go:367 +0x1890
sigs.k8s.io/controller-tools/pkg/loader.EachType(0xc00b6e23c0, 0xc00cc8f818)
        /root/go/pkg/mod/sigs.k8s.io/controller-tools@v0.8.0/pkg/loader/visit.go:38 +0x96
sigs.k8s.io/controller-tools/pkg/loader.allReferencedPackages(0xc00b6e23c0, 0xc00b6dc600)
        /root/go/pkg/mod/sigs.k8s.io/controller-tools@v0.8.0/pkg/loader/refs.go:162 +0x266
sigs.k8s.io/controller-tools/pkg/loader.(*TypeChecker).check(0xc0075d1b00, 0xc00b6e23c0)
        /root/go/pkg/mod/sigs.k8s.io/controller-tools@v0.8.0/pkg/loader/refs.go:249 +0x187
sigs.k8s.io/controller-tools/pkg/loader.(*TypeChecker).check.func1(0x47?)
        /root/go/pkg/mod/sigs.k8s.io/controller-tools@v0.8.0/pkg/loader/refs.go:257 +0x65
created by sigs.k8s.io/controller-tools/pkg/loader.(*TypeChecker).check
        /root/go/pkg/mod/sigs.k8s.io/controller-tools@v0.8.0/pkg/loader/refs.go:255 +0x1cd
tools/make/kube.mk:12: recipe for target 'generate' failed
```

**Comment by Xunzhuo** [comment]
<img width="872" alt="image" src="https://user-images.githubusercontent.com/48784001/188100581-0fbe29af-c781-47ea-ab24-63db82b1f81c.png">
I did not see anything wrong?

**Comment by zirain** [comment]
yes, pass on my mac-pro, but failed on my dev machine 

```
root@ecs-zirain:~# uname -a
Linux ecs-zirain 5.10.5-051005-generic #202101061537 SMP Wed Jan 6 15:43:53 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
root@ecs-zirain:~# lsb_release 
No LSB modules are available.
root@ecs-zirain:~# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.6 LTS
Release:	18.04
Codename:	bionic
```

**Comment by arkodg** [comment]
> yes, pass on my mac-pro, but failed on my dev machine
> 
> ```
> root@ecs-zirain:~# uname -a
> Linux ecs-zirain 5.10.5-051005-generic #202101061537 SMP Wed Jan 6 15:43:53 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
> root@ecs-zirain:~# lsb_release 
> No LSB modules are available.
> root@ecs-zirain:~# lsb_release -a
> No LSB modules are available.
> Distributor ID:	Ubuntu
> Description:	Ubuntu 18.04.6 LTS
> Release:	18.04
> Codename:	bionic
> ```

`go version` same on both machines ?

**Comment by zirain** [comment]
on ubuntu18
```
go version go1.18.2 linux/amd64
```

mac
```
go version go1.18.1 darwin/amd64
```

**Comment by hnlq715** [comment]
Same error on my ubuntu server.

It would be fixed when downgrade grpc to 1.45.0 with replace in gomod file.

```
replace google.golang.org/grpc => google.golang.org/grpc v1.45.0
```

**Comment by arkodg** [comment]
hey @hnlq715 @zirain thank for further debugging this issue
does bumping controller-runtime to [v0.13.0](https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.13.0) fix the issue ? if not lets downgrade `google.golang.org/grpc` to v1.45.0
relates to https://github.com/kubernetes-sigs/controller-tools/pull/648
